### ISSUE TITLE
Update for changes in minecraft 1.8

### ISSUE THREAD

**Issue by Peter200lx** [Peter200lx/ToolBelt#38]
This is a huge task, as 1.8 added/changed a bunch of stuff:
http://minecraft.gamepedia.com/1.8

Eventually should transition to Block States
http://minecraft.gamepedia.com/Block_states

### NBT Tags

These are interesting. Probably of no use, but worth investigating.

#### BlockEntityTag
        Block entities item form can hold custom NBT tags that are merged to the specified block entity when they're placed.
        Example: /give @p command_block 1 0 {BlockEntityTag:{Command:"/setblock ~ ~ ~ minecraft:diamond_block"}}
        Gives a command block that, when placed, will automatically have the command /setblock ~ ~ ~ minecraft:diamond_block set.

#### SelectedItem
        Contains the item data of the item currently held in the player's hand.
        Added to compact the process of testing if a player is holding a specific item.
            Previous method required nine commands (one per slot in the hotbar); now requires only one command.

## Block IDs

    Instances of blocks/items references in commands, can now be named ids
        Example: id:"minecraft:stone".
        In scoreboards, the format is minecraft.block.stone.
        Old numerical ids are still supported, but will eventually be phased out.
        This is part of the requirements for the Plugin API.

## Block handling

    Now uses block states internally instead of metadata.
    Metadata will still be used for a while.
    Metadata no longer needs to be calculated out of the 4-bit data value; instead, the values of specified properties can now be easily gotten and set.
        Example: .minecraft/saves/TheMapWithTheThing/resources.zip.

## Blocks changed

    1 	1 	minecraft:stone 	Stone S B
    3 	3 	minecraft:dirt 	Dirt S B
    19 	13 	minecraft:sponge 	Sponge S B
    23 	17 	minecraft:dispenser 	Dispenser S E
    48 	30 	minecraft:mossy_cobblestone 	Moss Stone
    64 	40 	minecraft:wooden_door 	Oak Door I S
    67 	43 	minecraft:stone_stairs 	Cobblestone Stairs S
    77 	4D 	minecraft:stone_button 	Stone Button S
    85 	55 	minecraft:fence 	Fence
    98 	62 	minecraft:stonebrick 	Stone Bricks S B
    107 	6B 	minecraft:fence_gate 	Fence Gate S
    137 	89 	minecraft:command_block 	Command Block E
    138 	8A 	minecraft:beacon 	Beacon E
    143 	8F 	minecraft:wooden_button 	Wooden Button S
    144 	90 	minecraft:skull 	Mob head I S E
    151 	97 	minecraft:daylight_detector 	Daylight Sensor E
    157 	9D 	minecraft:activator_rail 	Activator Rail S
    167 	A7 	minecraft:iron_trapdoor 	Iron Trapdoor S
    170 	AA 	minecraft:hay_block 	Hay Bale S
    174 	AE 	minecraft:packed_ice 	Packed Ice

#### minecraft:stone 1 1 (Diorite, Andesite, Granite)

Diorite - Crafting recipe: 2 cobblestone and 2 Nether quartz in a checker board pattern.
Andesite - Crafting recipe: 1 cobblestone and 1 diorite.
Granite - Crafting recipe: 1 Nether quartz and 1 diorite.
Polished variants - Crafting recipe: 4 pieces of one of the materials, in a 2×2 configuration.

 	DV 	Description
	0 	Stone
	1 	Granite
	2 	Polished Granite
	3 	Diorite
	4 	Polished Diorite
	5 	Andesite
	6 	Polished Andesite

#### minecraft:dirt 3 3 (Coarse dirt)

 	DV 	Description
	0 	Dirt
	1 	Coarse Dirt
	2 	Podzol

.

    Replaces the grassless dirt variant found in mega taiga, mesa and savanna biomes.
        All worlds with grassless dirt blocks will seamlessly change over to coarse dirt, as it uses the same block ID and data value as the original grassless dirt block.
    Slightly darker texture than regular dirt.
    Can be picked up with bare hands (silk touch is not required).
    Tilling coarse dirt with a hoe will turn it to regular dirt.
    Crafting recipe: dirt and gravel in a 2×2 checkered pattern yields four coarse dirt.

#### minecraft:sponge 19 13

 	DV 	Description
	0 	Sponge
	1 	Wet Sponge

    Reintroduced sponge to survival mode, with new behavior and texture.
    Sponge turns into wet sponge when it soaks up water.
        Water particles appear around the sponge when this happens.
    Sponge destroys water blocks from 5 blocks away in a kind of sphere.
    Sponge soaks up water, only when some water is touching it.

Wet sponge

    Obtained when a sponge soaks up water.
    Smelting a wet sponge yields a dry sponge.
        When smelted, they will fill empty buckets in the fuel slot with water if possible.
    Emits water dripping particles while placed.
    Dropped by elder guardians on player kills.

#### minecraft:dispenser 23 17

    Can place pumpkins or wither skeleton skulls to spawn golems and withers respectively.
        Will only place them if the body of the golem or wither is already built, in a valid configuration.
    Can place command blocks with pre-configured commands.

#### minecraft:mossy_cobblestone 48 30

    Now craftable from one cobblestone and one vine.

#### minecraft:wooden_door 64 40

    Stack to 64.
    The crafting recipe for doors gives three doors instead of one.
    Door item textures were given more detail, to be consistent with the item textures of the new doors.
        The above changes apply to both oak doors and iron doors.
    "Wooden Doors" have been renamed to "Oak Doors".
        This was made in order to accommodate for the new door variants.

#### minecraft:stone_stairs 67 43 (Cobblestone Stairs)

    "Stone Stairs" were renamed to "Cobblestone Stairs" to avoid ambiguity.

#### minecraft:stone_button 77 4D

    Can be placed on the ceiling and on the ground.

#### minecraft:stonebrick 98 62

Mossy - Now craftable from one stone brick and one vine.
Cracked - Can now be obtained by smelting stone bricks, making them renewable.
Chiseled - Now craftable from two stone brick slabs on top of each other.

#### minecraft:command_block 137 89

    Survival players cannot longer look inside the command block to see the command.
    Show the @e target selector description, along with the rest in the UI.
    Added an X/O button next to "last output" in the command block UI, to toggle the visibility of the last output string.

#### minecraft:beacon 138 8A

    Beacon light beams change color when going through stained glass and stained glass panes.
        Making it feed into multiple stained glass blocks/panes will combine the colors.
    Beam now goes through all blocks that don't completely block light: Ice, water and leaves.

#### minecraft:wooden_button 143 8F

    Can be placed on the ceiling and on the ground.

#### minecraft:skull 144 90

    Made creeper, skeleton and zombie heads available in survival.
        Creepers, skeletons, wither skeletons and zombies drop their heads when killed by charged creepers. Wither skeletons continue to have heads as rare drops.
        Charged creeper explosions will not yield more than one mob head, regardless of how many mobs were killed by it.
    Skulls worn on heads are bigger, so the 2nd skin layer doesn't render outside.
    Player heads now have a 3D item model.
    Player/mob heads, both in inventories and held by mobs/players, now display the actual head skin/texture-including the second skin layer of player skins.
        Placed player heads also render the 2nd skin layer.

#### minecraft:daylight_detector 151 97

    Can be inverted with a right click, which changes the color of the nine "sensors" to change to a blue hue.
        Reaches full strength at the opposite time than it would when set to normal.
        This can allow for a light that turns on at night rather than day, without the need of a NOT gate.

#### minecraft:activator_rail 157 9D

    Powered activator rails will shake minecarts, causing riders (players or mobs) to dismount.

#### minecraft:hay_block 170 AA

    Rebalanced how they heal and accelerate the growth of horses.
        Now do it nine times as fast.

#### minecraft:packed_ice 174 AE

    Mined faster with a pickaxe.

#### Fences, fence gates & trapdoors

    "Fences" and "Fence Gates" have been renamed to "Oak Fences" and "Oak Fence Gates" respectively.
    "Trapdoors" were renamed to "Wooden Trapdoors"
        The above changes were made to facilitate the distinction, with the new variants for each block.

.

    New types of fences, fence gates and doors, for all the different types of wood.
    Different types of fences and fence gates can connect to each other (except nether brick fences as usual).
    Every recipe requires all wood planks to be of the same type.
        New crafting recipe for fences: 2 sticks and 4 planks to make 3 fences.
    Different types of doors have different models, with some types having no windows.

## New Blocks added

    165 	A5 	minecraft:slime 	Slime Block
    166 	A6 	minecraft:barrier 	Barrier

    168 	A8 	minecraft:prismarine 	Prismarine S B
    169 	A9 	minecraft:sea_lantern 	Sea Lantern
    170 	AA 	minecraft:hay_block 	Hay Bale S
    171 	AB 	minecraft:carpet 	Carpet S B

    173 	AD 	minecraft:coal_block 	Block of Coal

    176 	B0 	minecraft:standing_banner 	Standing Banner I S E
    177 	B1 	minecraft:wall_banner 	Wall Banner I S E
    178 	B2 	minecraft:daylight_detector_inverted 	Inverted Daylight Sensor E
    179 	B3 	minecraft:red_sandstone 	Red Sandstone S B
    180 	B4 	minecraft:red_sandstone_stairs 	Red Sandstone Stairs S
    181 	B5 	minecraft:double_stone_slab2 	Double Red Sandstone Slab S
    182 	B6 	minecraft:stone_slab2 	Red Sandstone Slab S
    183 	B7 	minecraft:spruce_fence_gate 	Spruce Fence Gate
    184 	B8 	minecraft:birch_fence_gate 	Birch Fence Gate
    185 	B9 	minecraft:jungle_fence_gate 	Jungle Fence Gate
    186 	BA 	minecraft:dark_oak_fence_gate 	Dark Oak Fence Gate
    187 	BB 	minecraft:acacia_fence_gate 	Acacia Fence Gate
    188 	BC 	minecraft:spruce_fence 	Spruce Fence
    189 	BD 	minecraft:birch_fence 	Birch Fence
    190 	BE 	minecraft:jungle_fence 	Jungle Fence
    191 	BF 	minecraft:dark_oak_fence 	Dark Oak Fence
    192 	C0 	minecraft:acacia_fence 	Acacia Fence
    193 	C1 	minecraft:spruce_door 	Spruce Door I S
    194 	C2 	minecraft:birch_door 	Birch Door I S
    195 	C3 	minecraft:jungle_door 	Jungle Door I S
    196 	C4 	minecraft:acacia_door 	Acacia Door I S
    197 	C5 	minecraft:dark_oak_door 	Dark Oak Door I S

* _I = Has a different ID as an inventory item_
* _D = Use the item's Damage field to define its durability._
* _S = Requires additional data from the saved game's Data array to fully define the block._
* _B = Requires additional data in the item's Damage field to fully define the inventory item._
* _E = Requires a block entity to store additional data._

####  minecraft:slime 165 A5

    Crafting recipe: 9 slime balls.
        Can be crafted back into 9 slime balls.
    Players and mobs that land on their top side will bounce, like on a trampoline.
        This negates all fall damage.
            Holding ⇧ Shift will negate the rebound while still negating the fall damage.
        Bounce rebound velocity is scaled by impact velocity.
            Height can reach up to 60% of initial height, which implies a great restitution coefficient.
        Does not affect falling item entities.
    Walking on it is slower than walking on soul sand and close to one's speed while sneaking.
    The placing/breaking sounds use the hurt sounds of slimes.
    Can be mined with only one click, without any tools in survival.
    Slime blocks will push and pull each other, along with adjacent blocks, when at least one of them is moved by a sticky piston.
        Normal pistons will have the same effect, except that they fail to pull even a single slime block.
        Coded by KaboPC and Panda4994.[3][4]
            Follow all the existing rules that pistons followed:
                A chain of slime blocks and its adjacent blocks connected to a piston, in any arrangement, can be moved as long as the following conditions are met:
                    The piston is not moving more than twelve blocks at a time.
                    Non-movable blocks are not obstructing the path.
                    Not a single slime block in the chain is attached to the piston itself.
        Allows for moveable contraptions.
    Extending a piston with a slime block attached will propel adjacent entities (mobs, players, items, launched arrows, etc.) in the appropriate direction.[5]

####  minecraft:barrier 166 A6

    Acts like bedrock, but is completely transparent.
        Can transfer redstone signals and allows blocks to be placed on it.
        Can only be destroyed and obtained in Creative mode.
        It gives the same particles as lava does when destroyed.
    Becomes visible to players who are holding a barrier in their hand, while in creative.
        Appears to be rendered as specially big particles, in the form of a red and square cross-out symbol.
            There is exactly one particle inside of every block.
            The face always follows the player.
            That texture is also used for its item form.
            The particles are only rendered for the blocks in the vicinity of the player.

####  minecraft:iron_trapdoor 167 A7

Looks like the same as a wooden trapdoor.

    Can only be opened and closed using redstone, similar to an iron door.
    Crafting recipe: iron ingots in a 2×2 pattern yields one iron trap door.

####  minecraft:prismarine 168 A8

 	DV 	Description
	0 	Prismarine
	1 	Prismarine Bricks
	2 	Dark Prismarine

Prismarine
    Generates in ocean monuments.
    Can be crafted with prismarine shards.
    The cracks in prismarine appear to slowly change color between brown, blue, gray and purple.

Prismarine bricks
    Generate in ocean monuments.
    Can be crafted with prismarine shards.

Dark prismarine
    Generate in ocean monuments.
    Can be crafted with prismarine shards and an ink sac.

####  minecraft:sea_lantern 169 A9

    Generate in ocean monuments.
    Can be crafted with prismarine shards and prismarine crystals.
    Emit light at a light level of 15 and have a subtle animated texture.

#### minecraft:hay_block 170 AA

```
Name 	Value 	Description
 axis 	x 	The ends of the hay block point east-west.
 axis 	y 	The ends of the hay block point up-down.
 axis 	z 	The ends of the hay block point north-south.
```

#### minecraft:carpet & minecraft:stained_hardened_clay

Same data set as wool

####  minecraft:standing_banner 176 B0 & minecraft:wall_banner 177 B1

A four-bit field storing a value from 0 to 15 specifying the block's orientation:

Standing:

    0: south
    1: south-southwest
    2: southwest
    3: west-southwest
    4: west
    5: west-northwest
    6: northwest
    7: north-northwest
    8: north
    9: north-northeast
    10: northeast
    11: east-northeast
    12: east
    13: east-southeast
    14: southeast
    15: south-southeast

Wall:

0x8 is unused
A three-bit field storing a value from 2 to 5 specifying the block's orientation:

    2: north
    3: south
    4: west
    5: east

.

    Crafted like signs, but with one color of wool instead of planks.
    Can be stacked to 16.
    Most patterns are created by arranging dyes in certain ways around a banner on a crafting table (see the page on banners for a full list).
        Other patterns are achieved with specific items.
            Item – pattern: "creeper skull – creeper face", "wither skeleton skull – skull and crossbones", "brick block – brick texture background", "oxeye daisy – flower icon", "vines – curly border" and "enchanted golden apple – Mojang logo.
            Making these patterns without dyes will result in a black color by default
    Banners can have up to six layers (in survival).
        Over a quadrillion possible combinations, counting rotations and visually identical patterns due to occlusion.
        The layers show in the order they were created.
        Custom banners with more than six layers are possible using commands.
    Top-most layer can be removed using cauldrons.
        This removes a third of the water of a full cauldron.
        All layers can be removed.
    Can be placed on walls or on the ground.
    They are visually like an entity that is one block wide and two blocks tall, but in reality they are a block occupying the lower portion, but having a bigger model.
        The hitbox is smaller than a full block, and it lacks any collision properties.
        On the ground they have 16 different possible rotations (like an armor stand).
        Has an animation when placed on the ground, that simulates swaying with the wind.
    Can be worn as head gear, but only using the /replaceitem command.
    Can be cloned by crafting them together, with a blank banner of the corresponding base color.
    Can be used as fuel in furnaces.

####  minecraft:red_sandstone 179 B3

 	DV 	Description
	0 	Red Sandstone
	1 	Chiseled Red Sandstone
	2 	Smooth Red Sandstone

    Can be crafted with 4 red sand in a square.
    Can be crafted into chiseled, smooth, stair and slab forms.
        Chiseled variant has a wither pattern to complement the creeper pattern found in chiseled sandstone.
        Slabs have a smooth variant, similar to stone and sandstone.
        Comes as a smooth double slab – minecraft:double_stone_slab2:8.
    Generated only in Mesa biomes at cave entrances.

#### minecraft:red_sandstone_stairs 180 B4

Most likely the same as standard stairs

#### minecraft:double_stone_slab2 181 B5

 	DV 	Description
	0 	Double Red Sandstone Slab
	8 	Smooth Double Red Sandstone Slab

#### minecraft:stone_slab2 182 B6

Single slabs can be either "right-side-up" or "upside-down"; this information is stored in the most significant metadata bit 0x8 as follows:

    0: Slab is right-side-up, occupying the bottom half of its voxel.
    1: Slab is upside-down, occupying the top half of its voxel

## Blocks removed

None?

**Comment by Peter200lx** [comment]
Alright, I think (am hoping) that I only have the banners left to deal with for all of the block changes.

**Comment by Peter200lx** [comment]
I believe I have all of the block related changes. Ideally I'll now verify logging and region protection, but otherwise I'm hopefully ready for release.

**Comment by Peter200lx** [comment]
I've now gone through all of the tools and I think I have all of the blocks and corner cases covered. I think I've only got to update a bit of the help text before release.

**Comment by Peter200lx** [comment]
Fixed with release 0.11
